### PR TITLE
chore(entities): Subtype class names are no longer stored in the DB

### LINKED ISSUE #6802: varchar(50) for class column in entity_subtypes table is restrictive
As I am starting to use PSR-4 for autoloading entity classes, the callable class names are getting rather long and don't fit into the 50 chars allocated for class names.

### PR DESCRIPTION
**Pull request by mrclay**
This stores class names only in an in-memory cache. The “class” column is left in the DB for now but is not read.

BREAKING CHANGE:
add_subtype() must now be called in a plugin’s init function.

Fixes #6802
- [ ] make BC with 2.x
- [ ] make update_subtype() an alias for add_subtype()

### CODE REVIEW

**Comment by mrclay** [comment]
I didn't touch the tests yet. Might be simpler to just make the class column bigger for now.

**Comment by mrclay** [comment]
I think what's happening [here](https://travis-ci.org/Elgg/Elgg/jobs/69588839#L580) is that the installer has established a connection, but the SubtypeTable class is trying to create a connection through the core DB service.

**Comment by iionly** [comment]
Am I correct in my understanding that this will no longer be a BC breaking change included in 2.0 or could this still happen?

**Comment by hypeJunction** [comment]
Let's just increase the col width for now. I think it's a pandora box

File: engine/lib/entities.php [hunk 1]
```diff
@@ -268,16 +229,14 @@ function remove_subtype($type, $subtype) {
...
- * @param string $subtype Subtype
- * @param string $class   Class name to use when loading this entity
+ * @tip This function exists only to allow plugins to be BC with 1.9 and earlier
  *
  * @return bool
  */
-function update_subtype($type, $subtype, $class = '') {
-	return _elgg_services()->subtypeTable->update($type, $subtype, $class);
+function update_subtype() {
+	return true;
```
Comment (mrclay) [inline engine/lib/entities.php]: should rather be an alias for add_subtype()

**Comment by mrclay** [comment]
@hypeJunction I think I can make this BC for 2.x

**Comment by hypeJunction** [comment]
Let's do it. I want to get rid of stupid update_subtype(). Causes so much hassle in plugin development.

**Comment by hypeJunction** [comment]
Ref https://github.com/Elgg/Elgg/pull/10022/commits/dfcff728dd22cdc2055975f0279dd2a3aa8b4030#diff-20b56f78ca68ff21f308104b59cd0155R670

**Pull request closed without merging**